packages/compass-aggregations/src/components/output-stage-preview.tsx

```diff
@@ -0,0 +1,228 @@
+import type { ThunkDispatch } from "redux-thunk";
+
+import type { ExtraArgs, RootState } from '../modules';
+import { gotoOutResults } from '../modules/out-results-fn';
+import { runStage } from '../modules/pipeline-builder/stage-editor';
+import {
+  getDestinationNamespaceFromStage,
+  MERGE_STAGE_PREVIEW_TEXT,
+  OUT_STAGE_PREVIEW_TEXT,
+} from '../utils/stage';
+import { parseShellBSON } from '../modules/pipeline-builder/pipeline-parser/utils';
+
+const stagePreviewOutStyles = css({
+  display: 'flex',
+  flexDirection: 'column',
+  alignItems: 'center',
+  width: '100%',
+  paddingTop: spacing[5],
+  paddingLeft: spacing[1],
+  paddingRight: spacing[1],
+  paddingBottom: spacing[2],
+});
+
+const stagePreviewOutTextStyles = css({
+  padding: `0 ${spacing[3]}px ${spacing[1]}px ${spacing[3]}px`,
+  textAlign: 'center',
+  marginBottom: spacing[2]
+});
+
+const stagePreviewOutLinkStyles = css({
+  border: 'none',
+  padding: 0,
+  margin: 0,
+  background: 'none',
+});
+
+const loaderStyles = css({
+  display: 'flex',
+  alignItems: 'center',
+  gap: spacing[2],
+});
+
+type OutputStageProps = {
+  isLoading: boolean;
+  hasServerError: boolean;
+  isFinishedPersistingDocuments: boolean;
+  isAtlasDeployed: boolean;
+  destinationNamespace: string;
+  onRunOutputStage: () => void;
+  onGoToOutputResults: () => void;
+};
+
+const documentsPersistedText = (destination: string | null) => {
+  const location = destination
+    ? `collection: ${destination}`
+    : `specified collection`;
+  return `Documents persisted to ${location}`;
+}
+
+const Loader = ({
+  destinationNamespace,
+}: {
+  destinationNamespace: string | null;
+}) => {
+  return (
+    <div className={stagePreviewOutStyles}>
+      <div className={loaderStyles}>
+        <SpinLoader />
+        Persisting Documents {
+          destinationNamespace
+          ? `to ${destinationNamespace}`
+          : '...'
+        }
+      </div>
+    </div>
+  )
+};
+
+export const MergeStage = ({
+  isLoading,
+  hasServerError,
+  isFinishedPersistingDocuments,
+  isAtlasDeployed,
+  destinationNamespace,
+  onRunOutputStage,
+  onGoToOutputResults,
+}: OutputStageProps) => {
+
+  if (isLoading) {
+    return <Loader destinationNamespace={destinationNamespace} />;
+  }
+
+  // Stage editor show the error message.
+  if (hasServerError) {
+    return null;
+  }
+
+  if (isFinishedPersistingDocuments) {
+    return (
+      <div className={stagePreviewOutStyles}>
+        <Body className={stagePreviewOutTextStyles}>
+          {documentsPersistedText(destinationNamespace)}
+        </Body>
+        <Link
+          data-testid="go-to-merge-collection"
+          as="button"
+          className={stagePreviewOutLinkStyles}
+          onClick={onGoToOutputResults}
+        >
+          Go to collection.
+        </Link>
+      </div>
+    );
+  }
+
+  return (
+    <div className={stagePreviewOutStyles}>
+      <div className={stagePreviewOutTextStyles}>
+        {MERGE_STAGE_PREVIEW_TEXT}
```

Comment:
Is this the only difference between the two components in this file?

Reply:
Kind of. I didn't extract it to one component in order to avoid breaking e2e tests. I'll create a follow up for this to use same component for this.

Reply:
Created a follow up: [COMPASS-6451](https://jira.mongodb.org/browse/COMPASS-6451)